shimmy/utils/meltingpot.py

```diff
@@ -0,0 +1,80 @@
+"""Shared utils for meltingpot."""
+
+# pyright:  reportGeneralTypeIssues=false
+from typing import Any, Mapping, OrderedDict
+
+import dm_env
+import numpy as np
+import tree
+from gymnasium import spaces
+from pettingzoo.utils.env import ObsDict
+
+PLAYER_STR_FORMAT = "player_{index}"
+_WORLD_PREFIX = "WORLD."
+
+
+def dm_spec2gym_space(spec: tree.Structure[dm_env.specs.Array]) -> spaces.Space:
```

Comment:
Im still confused the difference between this function and the `utils.dm_env.dm_spec2gym_spec`. Why do we need this unique function

Reply:
Actually you have a point. I've just looked at it and they're the same. The only catch is that the other one doesn't have tuple. That said, I think I still prefer if both utilities are kept separate.

Reply:
I can look into this again but last time I had some issues merging them for some reason or other, might have been to do with the meltingpot CI tests though. Should be pretty straightforward to extend it to cover both use cases.

Reply:
Got this fixed